src/librustc/traits/project.rs

```diff
@@ -1247,6 +1247,34 @@ fn confirm_callable_candidate<'cx, 'gcx, 'tcx>(
                                               fn_sig,
                                               flag);
 
+    if selcx.projection_mode().is_any() {
+        // Issue #33364: `ProjectionMode::Any` means trans, during
+        // which there is no (good) reason for confirmation to fail
+        // (because type check has already admitted the source code).
```

Comment:
@eddyb Wait so are you saying it is not actually sound for me to skip the confirms via the logic I have put here?

Or ... are you "merely" saying that the comment as written here is misleading?